source/scene3.cpp

```diff
@@ -55,12 +55,15 @@ void Scene3::render(SDL_Renderer *renderer) {
     }
 #ifdef _XBOX
     case UpdateState::CheckFirmware: {
-      HalReadSMBusValue(I2C_HDMI_ADRESS, I2C_BOOT_MODE, false, &current_boot);
-
-      console->print("Found... Revision: 3C");
-      console->print("Firmware Mode: 0x%02X", current_boot);
+      if(HalReadSMBusValue(I2C_HDMI_ADRESS, I2C_BOOT_MODE, false, &current_boot) != 0) {
```

Comment:
Nit: in surrounding code, other comparisons against 0 just checks the value itself (without the `!= 0` part).

Reply:
Replicated the way the checks were performed here
https://github.com/MakeMHz/xbox-hdmi-app/blob/master/source/scene0.cpp

Reply:
```
if(HalReadSMBusValue(I2C_HDMI_ADRESS, I2C_FIRMWARE_VERSION + 0, false, &smbus_read) != 0) {
    snprintf(text_buffer, sizeof(text_buffer), "XboxHDMI not detected!");
  } else {
    firmware_version[0] = (uint8_t)smbus_read;

    HalReadSMBusValue(I2C_HDMI_ADRESS, I2C_FIRMWARE_VERSION + 1, false, &smbus_read);
    firmware_version[1] = (uint8_t)smbus_read;

    HalReadSMBusValue(I2C_HDMI_ADRESS, I2C_FIRMWARE_VERSION + 2, false, &smbus_read);
    firmware_version[2] = (uint8_t)smbus_read;

    // Firmware 1.0.0 will incorrectly report 0.0.0, so let's fix that.
    if(firmware_version[0] == 0 && firmware_version[1] == 0 && firmware_version[2] == 0) {
      firmware_version[0] = 1;
    }

    snprintf(text_buffer, sizeof(text_buffer), "Firmware Version: %u.%u.%u",
      firmware_version[0], firmware_version[1], firmware_version[2]);
  }
```

Reply:
Just as a comment, this codebase doesn't have a clear style, or standard. Things are done different ways in different places. I already copy/pasted the exact checks from a different file and I matched that style to "keep it the same" as much as possible.

In my humble opinion, if you want to nitpick, you should refactor the codebase and establish those standards. Otherwise, it will drive potential contributors away.

Reply:
Fair enough. I wasn't aware of the code you had copied when I made my remarks, I only compared to the surrounding code in the same file.
Apart from my small remarks (which you probably should ignore given your very reasonable explanation) your code looks good to me. 👍